File: src/main/java/com/cuupa/classificator/controller/ClassificatorController.java
package com.cuupa.classificator.controller;

import com.cuupa.classificator.gui.GuiProcess;
import com.cuupa.classificator.services.Classificator;
import com.cuupa.classificator.services.kb.semantic.SemanticResult;
import com.google.gson.Gson;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@Controller
public class ClassificatorController {

	private static final Gson gson = new Gson();

	private final Classificator classificator;

	public ClassificatorController(Classificator classificator) {
		this.classificator = classificator;
	}

	@RequestMapping("/ping")
	public ResponseEntity<String> ping() {
		return ResponseEntity.ok().body("200");
	}

	@RequestMapping(value = { "/", "/index" }, method = RequestMethod.GET)
	public String index(Model model) {
		model.addAttribute("guiProcess", new GuiProcess());
		return "index";
	}

	@RequestMapping(value = "/guiProcess", method = RequestMethod.POST)
	public String guiProcess(@ModelAttribute GuiProcess guiProcess) {
		List<SemanticResult> result = classificator.classify(guiProcess.getInputText());
		guiProcess.setResult(result);
		return "index";
	}

	@RequestMapping(value = "/classifyText", method = RequestMethod.POST)
	public ResponseEntity<String> classify(@RequestBody String text) {
		try {
			List<SemanticResult> result = classificator.classify(text);
			return ResponseEntity.status(HttpStatus.OK).body(gson.toJson(result));
		} catch (Exception e) {
			e.printStackTrace();
		}
		return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
	}

	@RequestMapping(value = "/classify", method = RequestMethod.POST)
	public ResponseEntity<String> classify(@RequestBody byte[] content) {
		try {
			List<SemanticResult> result = classificator.classify(content);
			return ResponseEntity.status(HttpStatus.OK).body(gson.toJson(result));
		} catch (Exception e) {
			e.printStackTrace();
		}
		return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
	}
}

File: src/main/java/com/cuupa/classificator/controller/ClassificatorController.kt
package com.cuupa.classificator.controller

import com.cuupa.classificator.knowledgebase.Classificator
import com.google.gson.GsonBuilder
import org.apache.commons.logging.Log
import org.apache.commons.logging.LogFactory
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class ClassificatorController(private val classificator: Classificator) {

    private val log: Log = LogFactory.getLog(ClassificatorController::class.java)

    @GetMapping(value = ["/api/rest/1.0/ping"])
    fun ping(): ResponseEntity<String> = ResponseEntity.ok().body("200")

    @PostMapping(value = ["/api/rest/1.0/classifyText"])
    fun classify(@RequestBody text: String?): ResponseEntity<String> {
        try {
            val result = classificator.classify(text)
            return ResponseEntity.status(HttpStatus.OK).body(gson.toJson(result))
        } catch (e: Exception) {
            log.error(e)
        }
        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null)
    }

    @PostMapping(value = ["/api/rest/1.0/classify"])
    fun classify(@RequestBody content: ByteArray?): ResponseEntity<String> {
        try {
            val result = classificator.classify(content)
            return ResponseEntity.status(HttpStatus.OK).body(gson.toJson(result))
        } catch (e: Exception) {
            log.error(e)
        }
        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null)
    }

    companion object {
        private val gson = GsonBuilder().serializeNulls().create()
    }
}